sql-parser/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/impl/PostgreSQLTCLStatementSQLVisitor.java

```diff
@@ -68,7 +71,40 @@ public PostgreSQLTCLStatementSQLVisitor(final Properties props) {
     
     @Override
     public ASTNode visitSetTransaction(final SetTransactionContext ctx) {
-        return new PostgreSQLSetTransactionStatement();
+        PostgreSQLSetTransactionStatement result = new PostgreSQLSetTransactionStatement();
+        if (null != ctx.transactionModeList()) {
+            ctx.transactionModeList().transactionModeItem().forEach(each -> {
+                setTransactionAccessType(result, each);
+                setTransactionIsolationLevel(result, each);
+            });
+        }
+        return result;
+    }
+    
+    private void setTransactionAccessType(final PostgreSQLSetTransactionStatement statement, final TransactionModeItemContext modeItemContext) {
+        TransactionAccessType accessType = null;
+        if (null != modeItemContext.ONLY()) {
+            accessType = TransactionAccessType.READ_ONLY;
+        } else if (null != modeItemContext.WRITE()) {
+            accessType = TransactionAccessType.READ_WRITE;
+        }
+        statement.setAccessMode(accessType);
+    }
+    
+    private void setTransactionIsolationLevel(final PostgreSQLSetTransactionStatement statement, final TransactionModeItemContext modeItemContext) {
```

Comment:
Can we refactor this method to getTransactionIsolationLevel and then use guard clause?

Reply:
OK.